turbinia/evidence.py

```diff
@@ -958,3 +958,38 @@ def _postprocess(self):
       # Unmount the container's filesystem
       mount_local.PostprocessUnmountPath(self._container_fs_path)
       self.state[EvidenceState.CONTAINER_MOUNTED] = False
+
+
+#TODO implement support for several ewf devices if there are more than one
+#inside the block path
+class EwfDisk(RawDisk):
+  """Evidence object for a EWF based evidence.
+
+  Attributes:
+    device_path (str): Path to the mounted loop device.
+    ewf_path (str): Path to mounted EWF image.
+    block_path (str): Path to EWF mount directory.
+  """
+  POSSIBLE_STATES = [EvidenceState.ATTACHED, EvidenceState.MOUNTED]
+
+  def __init__(self, *args, **kwargs):
+    """Initialization for EWF evidence object."""
+    self.device_path = None
+    self.ewf_path = None
+    self.block_path = None
+    super(EwfDisk, self).__init__(*args, **kwargs)
+
+  def _preprocess(self, _, required_states):
+    if EvidenceState.ATTACHED in required_states:
+      self.block_path = mount_local.PreprocessMountEwfDisk(self.source_path)
+      self.state[EvidenceState.MOUNTED] = True
```

Comment:
Ok, I looked a bit more into how ewfmount works, so I understand this better now.  I think for now we should just set ATTACHED to true (as you have below), and not set MOUNTED here because we're not actually mounting the block device yet (ie. actually exposing the disk file structure).    Then we can let the Partition enumeration task handle enumerating those and creating a DiskPartition evidence type that will actually set these to MOUNTED in its pre-processor.

@dfjxs Is there anything else we need to do here to make this new disk type work with the partition enumeration and mounting other than add it to the `evidence_intput` for the partition enumeration job?

Reply:
Yes, I agree. We should set ATTACHED and not MOUNTED.
Also, not a huge deal, but do we need an EWF path at all? Since the mounted EWF image is kind of a virtual block device, could we not just set the device_path to the mounted EWF rather than making another loopback device? The preprocessors further down the line are going to attach it anyway (and make another loopback device).

There shouldn't be anything else to do to have partition enumeration work though.

Reply:
I think it should actually be the `DiskPartition` pre-processors that get run after this and not the `RawDisk` ones.  Not all Tasks need the `DiskPartition` evidence type though (for example the Plaso task doesn't actually need to mount the partitions, it operates on the full disk).   Is it possible that you weren't seeing it being mounted because you were looking at the Tasks that didn't actually need it to be mounted?  

The `DiskPartition` pre-processor should also run the `PreprocessLosetup`, so I think @dfjxs is correct that we don't need to do that here.  I think the `MOUNTED` state still shouldn't be set here either (since that should also happen in the `DiskPartition` pre-processor.